File: workspace/2023-03-06/factorial.rust/factorial.rs

fn main()
{
    for n in 1..=10 {
        println!("factorial({}) = {}", n, factorial(n));
    }
}

fn factorial(n: i32) -> i32
{
    if n == 0 {
        1
    } else {
        n * factorial(n - 1)
    }
}

File: workspace/2023/2023-03-07/try-erlang.erlang/factorial.erl

main(_) ->
    demonstration(0),
    demonstration(1),
    demonstration(2),
    demonstration(3),
    demonstration(4),
    demonstration(5).

demonstration(N) ->
    io:format("factorial(~w) = ~w~n", [N, factorial(N)]).

factorial(0) -> 1;
factorial(N) -> N * factorial(N - 1).

